model.py

```diff
@@ -0,0 +1,31 @@
+from sklearn.compose import ColumnTransformer
+import tensorflow as tf
+import joblib
+import pandas as pd
+
+class Model:
+    def __init__(self) -> None:
+        pass
+    
+    def predict_on_batch(self, X: pd.DataFrame):
+        pass
+
+class TFModel(Model):
+    def __init__(self, tf_model_path: str, transformer_path: str) -> None:
+        Model.__init__(self)
+        self.vi_model =  self.load_tf_model(tf_model_path)
+        self.transformer = self.load_transformer(transformer_path)
+
+    def load_transformer(self, transformer_path: str) -> ColumnTransformer:
```

Comment:
Could we preface private/helper class methods here load_tf_model -> _load_tf_model so we know we shouldn't expect them to be called by the user?